### PR TITLE
Mtdsa 12257 Create a Historic FHL UK Property Income & Expenses Period Summary

### PR DESCRIPTION
**Pull request by abigail-houston-hmrc**

### CODE REVIEW

File: test/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncomeSpec.scala [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      |   }
+      |}
+      |""".stripMargin)
+
+  val desJson: JsValue = Json.parse(
+    """
+    |{
+    |    "rentIncome": {
+    |    "amount": 215.16,
+    |    "taxDeducted": 1365.12
```
Comment (sean-flynn-HMRC) [inline test/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncomeSpec.scala:44]: Could this json be formatted a bit more clearly?

File: app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncome.scala [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ * limitations under the License.
+ */
+
+package v2.models.request.common.ukFhlPieProperty
+
+import play.api.libs.functional.syntax.{toFunctionalBuilderOps, unlift}
+import play.api.libs.json.{JsPath, Reads, Writes, __}
+import v2.models.request.common.ukPropertyRentARoom.UkPropertyIncomeRentARoom
+
+case class UkFhlPiePropertyIncome(periodAmount: Option[BigDecimal], taxDeducted: Option[BigDecimal], ukRentARoom: Option[UkPropertyIncomeRentARoom])
```
Comment (sean-flynn-HMRC) [inline app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncome.scala:23]: If we are using the Pie acronym perhaps these model names could be even shorter? Maybe "UkFhlPieIncome"?

File: app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import v2.models.request.common.ukPropertyRentARoom.UkPropertyExpensesRentARoom
+
+
+case class UkFhlPiePropertyExpenses(premisesRunningCosts: Option[BigDecimal],
+                                 repairsAndMaintenance: Option[BigDecimal],
+                                 financialCosts: Option[BigDecimal],
+                                 professionalFees: Option[BigDecimal],
+                                 costOfServices: Option[BigDecimal],
+                                 other: Option[BigDecimal],
+                                 consolidatedExpenses: Option[BigDecimal],
```
Comment (sean-flynn-HMRC) [inline app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala:30]: It shouldn't be possible for vendors to submit a request containing both "consolidatedExpenses" and all the other expenses. 

Is the plan to allow consolidatedExpenses in the model and then later use the validator to invalidate requests that try to include consolidatedExpenses with the other expenses? 

Or would it make more sense to remove consolidatedExpenses from this model and have a separate model for UkFHLPiePropertyExpenses requests consisting of consolidatedExpenses only ?

File: app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+package v2.models.request.common.ukFhlPieProperty
+
+import play.api.libs.functional.syntax.{toFunctionalBuilderOps, unlift}
+import play.api.libs.json.{JsPath, Json, Reads, Writes}
+import v2.models.request.common.ukPropertyRentARoom.UkPropertyExpensesRentARoom
+
+
+case class UkFhlPiePropertyExpenses(premisesRunningCosts: Option[BigDecimal],
```
Comment (sean-flynn-HMRC) [inline app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala:24]: If we are using the Pie acronym perhaps these model names could be even shorter? Maybe "UkFhlPieExpenses"?

File: test/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncomeSpec.scala (hunk 1 above)
Comment (abigail-houston-hmrc) [reply test/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncomeSpec.scala:44]: absolutely

File: app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncome.scala (hunk 2 above)
Comment (abigail-houston-hmrc) [reply app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyIncome.scala:23]: Absolutely

File: app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala (hunk 3 above)
Comment (abigail-houston-hmrc) [reply app/v2/models/request/common/ukFhlPieProperty/UkFhlPiePropertyExpenses.scala]: Yes - model tests are only to ensure that mapping of model to and from json are successful

**Pull request merged**